### PR TITLE
add autobuild workflows

### PR DESCRIPTION
**Pull request by jaimergp**
Copied over from `conda/conda`. Using my own channel for now while we figure out the token policies across organizations.

Not sure if the action used allows to configure channels. I am using `conda_build_config.yaml` for now but I think that's a `conda-smithy` thing?

### CODE REVIEW

File: .github/workflows/builds-review.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            subdir: osx-64
+          - runner: windows-latest
+            subdir: win-64
+    runs-on: ${{ matrix.runner }}
+    steps:
+
+      - name: Remove build label
+        uses: actions/github-script@v5
+        with:
+          github-token: ${{ secrets.CANARY_ACTION_TOKEN }}
```
Comment (jaimergp) [inline .github/workflows/builds-review.yaml:28]: @jezdez which kind of token do we need here? Just a PAT for this repo?

Comment (jezdez) [reply .github/workflows/builds-review.yaml:28]: I'll add one for @conda-bot

Comment (jezdez) [reply .github/workflows/builds-review.yaml:28]: Done

File: .github/workflows/builds-review.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      # Clean checkout of specific git ref needed for package metadata version
+      # which needs env vars GIT_DESCRIBE_TAG and GIT_BUILD_STR:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.ref }}
+          clean: true
+          fetch-depth: 0
+
+      - name: Create and upload review build
+        uses: conda/actions/canary-release@v22.1.4
```
Comment (jaimergp) [inline .github/workflows/builds-review.yaml:53]:
```suggestion
        uses: conda/actions/canary-release@v22.2.0
```

File: .github/workflows/builds-canary.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      # Clean checkout of specific git ref needed for package metadata version
+      # which needs env vars GIT_DESCRIBE_TAG and GIT_BUILD_STR:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.ref }}
+          clean: true
+          fetch-depth: 0
+
+      - name: Create and upload canary build
+        uses: conda/actions/canary-release@v22.1.4
```
Comment (jaimergp) [inline .github/workflows/builds-canary.yaml:35]:
```suggestion
        uses: conda/actions/canary-release@v22.2.0
```

**Comment by jaimergp** [comment]
I don't think the token has the right permissions @jezdez:

```
RequestError [HttpError]: Must have admin rights to Repository.
    at /home/runner/work/_actions/actions/github-script/v5/dist/index.js:4378:21
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
Error: Unhandled error: HttpError: Must have admin rights to Repository.
    at async eval (eval at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v5/dist/index.js:4706:56), <anonymous>:11:5)
    at async main (/home/runner/work/_actions/actions/github-script/v5/dist/index.js:4761:20) {
  name: 'HttpError',
  status: 403,
  response: {
    url: 'https://api.github.com/repos/conda-incubator/conda-libmamba-solver/issues/5/labels/build%3A%3Areview',
    status: 403,
    headers: {
      'access-control-allow-origin': '*',
      'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset',
      connection: 'close',
      'content-encoding': 'gzip',
      'content-security-policy': "default-src 'none'",
      'content-type': 'application/json; charset=utf-8',
      date: 'Mon, 07 Feb 2022 20:31:33 GMT',
      'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
      server: 'GitHub.com',
      'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
      'transfer-encoding': 'chunked',
      vary: 'Accept-Encoding, Accept, X-Requested-With',
      'x-accepted-oauth-scopes': '',
      'x-content-type-options': 'nosniff',
      'x-frame-options': 'deny',
      'x-github-media-type': 'github.v3',
      'x-github-request-id': '07C6:38CB:2CA2FD:708940:620181A5',
      'x-oauth-scopes': 'repo',
      'x-ratelimit-limit': '5000',
      'x-ratelimit-remaining': '4996',
      'x-ratelimit-reset': '1644[26](https://github.com/conda-incubator/conda-libmamba-solver/runs/5099353885?check_suite_focus=true#step:2:26)9492',
      'x-ratelimit-resource': 'core',
      'x-ratelimit-used': '4',
      'x-xss-protection': '0'
    },
    data: {
      message: 'Must have admin rights to Repository.',
      documentation_url: 'https://docs.github.com/rest/reference/issues#remove-a-label-from-an-issue'
    }
  },
  request: {
    method: 'DELETE',
    url: 'https://api.github.com/repos/conda-incubator/conda-libmamba-solver/issues/5/labels/build%3A%3Areview',
    headers: {
      accept: 'application/vnd.github.-preview+json',
      'user-agent': 'actions/github-script octokit-core.js/3.5.1 Node.js/12.13.1 (linux; x[64](https://github.com/conda-incubator/conda-libmamba-solver/runs/5099353885?check_suite_focus=true#step:2:64))',
      authorization: 'token [REDACTED]',
      'content-length': 0
    },
    request: { agent: [Agent], hook: [Function: bound bound register] }
  }
}
```

**Comment by jezdez** [comment]
Oh thanks, I'll double-check

**Comment by jezdez** [comment]
Okay, I've added conda-bot to the conda-libmamba-solver team which should give it enough permissions.

**Pull request merged**